R/iterative_solver.R

```diff
@@ -38,7 +38,7 @@ solve_final_size_iterative <- function(contact_matrix,
   epi_final_size[i_here] <- 0.0
 
   # matrix filled by columns
-  contact_matrix_ <- contact_matrix * demography_vector %o% susceptibility
```

Comment:
That's a great question, which I don't have the theoretical background to answer. I was also surprised to find that the contact matrix filling differs between the Newton and iterative solvers in Edwin's C++ code, see
1. Iterative solver (only multiplying cols by demography): https://gitlab.com/epidemics-r/code_snippets/-/blob/feature/newton_solver/include/finalsize.hpp#L51
2. Newton solver (multiplying rows by susceptibility, and cols by demography): https://gitlab.com/epidemics-r/code_snippets/-/blob/feature/newton_solver/include/finalsize.hpp#L144

Initially, I thought that this must be a mistake, so I implemented both to be the same (taking the Newton solver as 'correct'). While fixing #69 I decided to go with what Edwin had written in case _I_ was wrong (about him making a mistake), and returned it to his implementation. I tried both implementations, and they both pass the solver equivalence tests in both R and C++, even with differences among susceptibility groups, which should make some difference. I have no idea why.